kod/object/active/holder/nomoveon/battler/player/user.kod

```diff
@@ -3749,6 +3758,44 @@ messages:
       return;
    }
 
+   UserMoveInventoryItem(payload=$,target=$)
+   % User moved an item in their inventory, now update plPassive list
+   {
+      local iPayload, iTarget;
+
+      % Sanity checks
+      if payload = $
+         OR target = $
+         OR payload = target
+         OR plPassive = $
+      {
+         return;
+      }
+
+      % Find the list positions for the two objects that the player wants to move
+      iPayload = FindListElem(plPassive,payload); 
+      iTarget = FindListElem(plPassive,target);
+
+      if iTarget > iPayload
+         {
+            iTarget = iTarget + 1;
```

Comment:
Something doesn't feel right here.  I believe this is needed because the client displays the inventory in reverse order?  But the Blakod doesn't care, and shouldn't know, the order in which the client displays items.  If this adjustment is necessary, it should be done on the client side.

Future readers are guaranteed to be confused why this is here.  When it's moved to the client, make sure to document it there.

Reply:
Removed this from user.kod, and made it client side in inventry.c

Client now supplies positional arguments in Blakod format to the server.  Documented this in comments.